src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1523,17 +1523,68 @@ public override SyntaxNode WithAccessibility(SyntaxNode declaration, Accessibili
+            DeclarationModifiers.Const |
+            DeclarationModifiers.New |
+            DeclarationModifiers.ReadOnly |
+            DeclarationModifiers.Static;
+        private static readonly DeclarationModifiers s_methodModifiers =
+            DeclarationModifiers.Abstract |
+            DeclarationModifiers.Async |
+            DeclarationModifiers.New |
+            DeclarationModifiers.Override |
+            DeclarationModifiers.Partial |
+            DeclarationModifiers.ReadOnly |
+            DeclarationModifiers.Sealed |
+            DeclarationModifiers.Static |
+            DeclarationModifiers.Virtual;
+        private static readonly DeclarationModifiers s_constructorModifiers =
+            DeclarationModifiers.Static;
+        private static readonly DeclarationModifiers s_propertyModifiers =
+            DeclarationModifiers.Abstract |
+            DeclarationModifiers.New |
+            DeclarationModifiers.Override |
+            DeclarationModifiers.ReadOnly |
+            DeclarationModifiers.Sealed |
+            DeclarationModifiers.Static |
+            DeclarationModifiers.Virtual;
+        private static readonly DeclarationModifiers s_eventModifiers =
+            DeclarationModifiers.Abstract |
+            DeclarationModifiers.New |
+            DeclarationModifiers.Override |
+            DeclarationModifiers.Sealed |
+            DeclarationModifiers.Static |
+            DeclarationModifiers.Virtual;
+        private static readonly DeclarationModifiers s_eventFieldModifiers =
+            DeclarationModifiers.New |
+            DeclarationModifiers.Static;
+        private static readonly DeclarationModifiers s_indexerModifiers =
+            DeclarationModifiers.Abstract |
+            DeclarationModifiers.New |
+            DeclarationModifiers.Override |
+            DeclarationModifiers.ReadOnly |
+            DeclarationModifiers.Sealed |
+            DeclarationModifiers.Virtual;
+        private static readonly DeclarationModifiers s_classModifiers =
+            DeclarationModifiers.Abstract |
+            DeclarationModifiers.New |
+            DeclarationModifiers.Partial |
+            DeclarationModifiers.Sealed |
+            DeclarationModifiers.Static;
+        private static readonly DeclarationModifiers s_structModifiers =
+            DeclarationModifiers.New |
+            DeclarationModifiers.Partial |
+            DeclarationModifiers.ReadOnly;
+        private static readonly DeclarationModifiers s_interfaceModifiers =
+            DeclarationModifiers.New |
+            DeclarationModifiers.Partial;
+        private static readonly DeclarationModifiers s_accessorModifiers =
+            DeclarationModifiers.New;
+        private static readonly DeclarationModifiers s_localDeclarationModifiers =
+            DeclarationModifiers.Ref;
+        private static readonly DeclarationModifiers s_parameterModifiers =
+            DeclarationModifiers.Ref;
```

Comment:
Looking at [DeclarationModifiers](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Editing/DeclarationModifiers.cs,63) and its properties, it seems it is designed only for top level declarations - i.e., namespaces, types, methods, fields, properties, and events, but not for other symbols such as locals or parameters. Its not good that there are absolutely no comments or even doc comments on the type, especially given its a public API.

Reply:
If we are planning to extend `DeclarationModifiers` to now include modifiers for non-top level declarations, then we need to extend it to include modifiers such as `out`, `in`, `volatilte`, etc. - we need a more exhaustive list. Additionally, this needs to go through a design meeting as we are extending the scope of an existing public API (although completely undocumented).

Reply:
> If we are planning to extend DeclarationModifiers to now include modifiers for non-top level declarations

We shoudl ask @mattwar .  But my remembrance is that this API was intended for use for all the use-cases of SyntaxGenerator.  In that regard, it would be expected to be meaningful on all these other constructs as well.  It's just the case that we often have shipped stuff that hit the 95% case, and allowed the rest to come in through bug fixes as features have needed additional functionality.

Reply:
Seems reasonable to extend - I'd still prefer it go through a design meeting and also we add the exhaustive list at least for new symbol kinds we are adding (parameters, locals). @chborl once finalize, can you please add doc comments to all the public APIs in this type.

Reply:
> I'd still prefer it go through a design meeting

Totally agree.  Public API.  So it def should be discussed.